compiler/test/suites/enums.re

```diff
@@ -40,4 +41,65 @@ describe("enums", ({test, testSkip}) => {
     recursive_contents,
     "Cons(Node(\"tree 1\", Cons(Node(\"tree 2\", Nil), Nil)), Nil)\n",
   );
+  // inline record variants
+  assertRun(
+    "enum_inline_record_1",
+    {|
+      enum Rec<a> {
+        Rec { x: a, y: a },
+        Tup(Number, Number, Number)
+      }
+      let r = Rec { x: 1, y: 2 }
+      let x = 11
+      let y = 12
+      let r2 = Rec { y, x }
+      print(r)
+      print(r2)
+      print(r == Rec { x: 1, y: 2 })
+      print(r == Rec { x: 2, y: 2 })
+      print(Tup(1, 2, 3))
+    |},
+    "Rec {\n  x: 1,\n  y: 2\n}\nRec {\n  x: 11,\n  y: 12\n}\ntrue\nfalse\nTup(1, 2, 3)\n",
+  );
+  assertCompileError(
+    "enum_inline_record_2",
+    {|
+      enum Rec<a> {
+        Rec { x: a, y: a }
+      }
+      let r = { x: 1, y: 2, }
+    |},
+    "Unbound record label x. Perhaps you need to import its type or write a type definition?",
```

Comment:
This is also the error message given for regular records. Should I try to fix it in this PR or should a separate issue be created?

Reply:
Prior to inline records this made sense because if the label is unknown, you'd either need to import a type or make a new record type. The error is entirely incorrect if I've written an example like you've written in the test, so we need to fix it for inline records specifically.

Reply:
Oh ok I see; I'll take a look then

Reply:
Is your suggestion to make it so that if the label exists on an inline record then the error could be "Unbound record label x. However, this label exists on inline record constructor Rec, which is not a standalone record definition"?

Reply:
That seems like a pretty good error! Is it hard to implement?

Reply:
I ended up implementing this but the code can probably be improved though it seems to work in all of the situations I tried.